### PR TITLE
Adding glouonts.torch.piecewise_linear

### PR DESCRIPTION
**Pull request by KelvinKan**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Please tag this pr with at least one of these labels to make our release process faster:** BREAKING, new feature, bug fix, other change, dev setup

### CODE REVIEW

File: src/gluonts/torch/distributions/piecewise_linear.py [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+
+        Parameters
+        ----------
+        z
+            observations, shape = gamma.shape = (*batch_size)
+
+        Returns
+        -------
+        torch.Tensor
+            u_tilde
```
Comment (lostella) [inline src/gluonts/torch/distributions/piecewise_linear.py:140]: Better to use a description:

```suggestion
            The quantile levels corresponding to the input observations.
```

File: src/gluonts/torch/distributions/piecewise_linear.py [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+        quantile = gamma + torch.sum(b * u_spline, dim=-1)
+
+        return quantile
+
+    def quantile(self, u: torch.Tensor) -> torch.Tensor:
+        return self.quantile_internal(u, dim=0)
+
+    def cdf(self, z: torch.Tensor) -> torch.Tensor:
+        r"""
+        compute the quantile levels u_tilde s.t. quantile(u_tilde) = z
```
Comment (lostella) [inline src/gluonts/torch/distributions/piecewise_linear.py:130]:
```suggestion
        Compute the quantile levels associated with the given observations.
```

File: src/gluonts/torch/distributions/piecewise_linear.py [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+
+        Parameters
+        ----------
+        z
+            observations, shape = gamma.shape = (*batch_size)
```
Comment (lostella) [inline src/gluonts/torch/distributions/piecewise_linear.py:135]:
```suggestion
            Observations, shape = gamma.shape = (*batch_shape)
```

File: src/gluonts/torch/distributions/piecewise_linear.py [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+        u_spline = F.relu(u - knot_positions)
+        quantile = gamma + torch.sum(b * u_spline, dim=-1)
+
+        return quantile
+
+    def quantile(self, u: torch.Tensor) -> torch.Tensor:
+        return self.quantile_internal(u, dim=0)
+
+    def cdf(self, z: torch.Tensor) -> torch.Tensor:
+        r"""
```
Comment (lostella) [inline src/gluonts/torch/distributions/piecewise_linear.py:129]: On a second thought, this docstring is not really needed, since this is the implementation of a method already documented in pytorch probably (we don't need to repeat this for all distributions). Just removing it should also fix the docs build

**Review by lostella — CHANGES_REQUESTED** [review]

**Review by lostella — APPROVED** [review]
Thanks @KelvinKan this looks good!
